### PR TITLE
Various refactoring of `CMapImages`

### PR DESCRIPTION
**Pull request by Robyt3**
## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/mapimages.cpp [hunk 1]
```diff
@@ -158,131 +158,133 @@ void CMapImages::LoadBackground(class CLayers *pLayers, class IMap *pMap)
...
+			for(int LayerType = 0; LayerType < MAP_IMAGE_ENTITY_LAYER_TYPE_COUNT; ++LayerType)
 			{
-				dbg_assert(!m_aaEntitiesTextures[(EntitiesModType * 2) + (int)EntitiesAreMasked][n].IsValid(), "entities texture already loaded when it should not be");
+				dbg_assert(!m_aaEntitiesTextures[(EntitiesModType * 2) + (int)EntitiesAreMasked][LayerType].IsValid(), "entities texture already loaded when it should not be");
 
 				// set everything transparent
 				mem_zero(pBuildImgData, BuildImageSize);
 
-				for(int i = 0; i < 256; ++i)
+				for(int i = TILE_AIR; i < 256; ++i)
```
Comment (heinrich5991) [inline src/game/client/components/mapimages.cpp:269]:
```suggestion
				for(int i = 0; i < 256; ++i)
```
The change made the code unclearer to me. It's supposed to iterate over all possible tile values, no?

File: src/game/client/components/mapimages.cpp [hunk 2]
```diff
@@ -260,17 +258,18 @@ IGraphics::CTextureHandle CMapImages::GetEntities(EMapImageEntityLayerType Entit
...
-						Graphics()->CopyTextureBufferSub(pBuildImgData, pTmpImgData, ImgInfo.m_Width, ImgInfo.m_Height, PixelSize, (size_t)X * CopyWidth, (size_t)Y * CopyHeight, CopyWidth, CopyHeight);
+						if(LayerType == MAP_IMAGE_ENTITY_LAYER_TYPE_SWITCH && TileIndex == TILE_SWITCHTIMEDOPEN)
+						{
+							TileIndex = 8;
+						}
+
+						const size_t CopyWidth = ImgInfo.m_Width / 16;
+						const size_t CopyHeight = ImgInfo.m_Height / 16;
+						const size_t OffsetX = (size_t)(TileIndex % 16) * CopyWidth;
+						const size_t OffsetY = (size_t)(TileIndex / 16) * CopyHeight;
```
Comment (heinrich5991) [inline src/game/client/components/mapimages.cpp:282]: It doesn't seem useful to let this type vary by CPU bit width, `uint32_t` would probably be more sensible. I see however that you pass it to a function that expects `size_t`, so I guess it's fine.

**Review by heinrich5991 — APPROVED** [review]
Looks good to me, only minor quibbles.

File: src/game/client/components/mapimages.cpp (hunk 1 above)
Comment (Robyt3) [reply src/game/client/components/mapimages.cpp]: Done.

**Review by heinrich5991 — APPROVED** [review]
Thanks!

**Pull request merged**